lib/connection/connection.js

```diff
@@ -422,7 +422,7 @@ function Connection(context) {
    * @returns {String} the query status.
    */
   this.getQueryStatusThrowIfError = async function (queryId) {
-    const status = this.getQueryStatus(queryId);
+    const status = await this.getQueryStatus(queryId);
```

Comment:
@sfc-gh-ext-simba-jy could you add test which show that it is working correctly now?

Reply:
I fixed the integration test cases.

The test cases in the unit test(snowflake_test) are working correctly because the driver throws an error on the 'getQueryResponse' function layer. As a result, it is hard to test 'getQueryStatusThrowIfError' with the unit test.

In the integration test (testExecuteAsync), the main reason why the testing was not working correctly was because of the 'assert.fail()' function. This function always throws an error whatever the results are, so this testing was not working appropriately. I edited this part to check that the driver throws the appropriate error. Please have a look at https://github.com/snowflakedb/snowflake-connector-nodejs/pull/746/files#diff-3399bc7071cb0f9829e2838ddaa9287958760a79f2a2f85d5074f3cadc8bcc5bL148